async_test.go

```diff
@@ -196,3 +196,63 @@ func TestLongRunningAsyncQuery(t *testing.T) {
 		}
 	}
 }
+
+func runLongRunningAsyncQuery(ctx context.Context, t *testing.T) {
+	runDBTest(t, func(dbt *DBTest) {
+		_ = dbt.mustQueryContext(ctx, "CALL SYSTEM$WAIT(100, 'SECONDS')")
+	})
+}
+
+func TestLongRunningAsyncQueryFetchResultByID(t *testing.T) {
+	queryIDChan := make(chan string, 1)
+	ctx := WithAsyncMode(context.Background())
+	ctx = WithQueryIDChan(ctx, queryIDChan)
```

Comment:
A lot of non GO SQL standard features are shown here! Maybe you can extend cmd/async/async.go with some demo and doc.go with description of this contexts?

Reply:
Which API do you want me to demo? cmd/async/async.go already demo for WithAsyncMode() and I think it's better to keep it simple as is.

Reply:
I meant fetching results by query id especially (`WithFetchResultByID`). It is worth extending the doc at least, with some code snippet.

Reply:
Added description in doc.go and an example cmd/fetchresultbyid/fetchresultbyid.go